pkg/ingester/pyroscope/ingest_adapter_test.go

```diff
@@ -0,0 +1,121 @@
+package pyroscope
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	pushv1 "github.com/grafana/pyroscope/api/gen/proto/go/push/v1"
+	typesv1 "github.com/grafana/pyroscope/api/gen/proto/go/types/v1"
+	"github.com/grafana/pyroscope/api/model/labelset"
+	"github.com/grafana/pyroscope/pkg/distributor/model"
+	"github.com/grafana/pyroscope/pkg/og/storage"
+	"github.com/grafana/pyroscope/pkg/og/storage/tree"
+
+	"connectrpc.com/connect"
+	"github.com/go-kit/log"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type MockLabelCapturingPushService struct {
```

Comment:
Just a note, not a suggestion: you may want to consider using mockery (see `.mockery.yaml`) to generate mocks and assertions in a more convenient way. (I agree that hand-written mocks can be more convenient in certain cases – it's totally up to you)

Reply:
Done in 6365cb043a942347c1ddf75570cbae500cd9edba.